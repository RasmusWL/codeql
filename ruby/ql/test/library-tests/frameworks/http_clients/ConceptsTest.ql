import ruby
import codeql.ruby.Concepts
import codeql.ruby.DataFlow
import TestUtilities.InlineExpectationsTest

// adopted from Python version in
// https://github.com/github/codeql/blob/ee45e799481356e0a59164f42ac8e4558aa09e61/python/ql/test/experimental/meta/ConceptsTest.qll#L515-L540

class HttpClientRequestTest extends InlineExpectationsTest {
  HttpClientRequestTest() { this = "HttpClientRequestTest" }

  override string getARelevantTag() {
    result in ["clientRequestUrlPart", "clientRequestCertValidationDisabled"]
  }

  override predicate hasActualResult(Location location, string element, string tag, string value) {
    exists(location.getFile().getRelativePath()) and
    exists(HTTP::Client::Request req, DataFlow::Node url |
      url = req.getAUrlPart() and
      location = url.getLocation() and
      element = url.toString() and
      value = url.toString() and
      tag = "clientRequestUrlPart"
    )
    or
    exists(location.getFile().getRelativePath()) and
    exists(HTTP::Client::Request req |
      req.disablesCertificateValidation(_) and
      location = req.getLocation() and
      element = req.toString() and
      value = "" and
      tag = "clientRequestCertValidationDisabled"
    )
  }
}
