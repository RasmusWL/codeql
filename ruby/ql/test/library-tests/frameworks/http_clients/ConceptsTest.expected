| Excon.rb:53:34:53:71 | # $ SPURIOUS: clientRequestUrlPart=url | Fixed spurious result:clientRequestUrlPart=url |
| Faraday.rb:3:21:3:41 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:6:22:6:42 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:9:21:9:41 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:12:23:12:43 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:15:24:15:44 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:18:22:18:42 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:24:23:24:43 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Faraday.rb:27:26:27:45 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| Faraday.rb:28:24:28:26 | "/" | Unexpected result: clientRequestUrlPart="/" |
| Faraday.rb:31:26:31:31 | "/foo" | Unexpected result: clientRequestUrlPart="/foo" |
| Faraday.rb:34:26:34:50 | Pair | Unexpected result: clientRequestUrlPart=Pair |
| Faraday.rb:34:31:34:50 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| Faraday.rb:35:25:35:27 | "/" | Unexpected result: clientRequestUrlPart="/" |
| HttpClient.rb:3:24:3:44 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:6:25:6:45 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:9:24:9:44 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:12:27:12:47 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:15:25:15:45 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:18:28:18:48 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:21:26:21:46 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:24:32:24:52 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| HttpClient.rb:26:34:26:54 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:5:14:5:34 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:7:15:7:35 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:9:14:9:34 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:11:16:11:36 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:15:25:15:45 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:18:23:18:43 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Httparty.rb:21:26:21:46 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| NetHttp.rb:4:15:4:17 | uri | Unexpected result: clientRequestUrlPart=uri |
| NetHttp.rb:6:23:6:36 | call to parse | Unexpected result: clientRequestUrlPart=call to parse |
| NetHttp.rb:13:14:13:16 | "/" | Unexpected result: clientRequestUrlPart="/" |
| NetHttp.rb:14:15:14:17 | "/" | Unexpected result: clientRequestUrlPart="/" |
| NetHttp.rb:15:14:15:16 | "/" | Unexpected result: clientRequestUrlPart="/" |
| NetHttp.rb:24:29:24:32 | path | Unexpected result: clientRequestUrlPart=path |
| OpenURI.rb:3:21:3:40 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| OpenURI.rb:6:14:6:33 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| OpenURI.rb:9:18:9:37 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| OpenURI.rb:15:26:15:46 | "https://example.com" | Unexpected result: clientRequestUrlPart="https://example.com" |
| RestClient.rb:3:24:3:44 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| RestClient.rb:6:25:6:44 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:9:24:9:43 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:12:26:12:45 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:15:27:15:46 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:18:25:18:44 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:21:28:21:47 | "http://example.com" | Unexpected result: clientRequestUrlPart="http://example.com" |
| RestClient.rb:28:56:28:84 | "http://example.com/resource" | Unexpected result: clientRequestUrlPart="http://example.com/resource" |
| Typhoeus.rb:3:22:3:42 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:6:23:6:43 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:9:22:9:42 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:12:24:12:44 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:15:25:15:45 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:18:23:18:43 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
| Typhoeus.rb:21:26:21:46 | "http://example.com/" | Unexpected result: clientRequestUrlPart="http://example.com/" |
